src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -44,6 +44,15 @@ private DrivetrainSubsystem() {
         });
 
     kill();
+
+    driveConfig.setUpdateCallback(
+        () -> {
+          Falcon500[] motors = getAllMotors().getMotorList();
+          for (Falcon500 motor : motors) {
+            motor.setPIDF(
+                driveConfig.getP(), driveConfig.getI(), driveConfig.getD(), driveConfig.getF());
+          }
+        });
```

Comment:
Could you explain this callback? If I understand correctly, it instantiates a Falcon500 motor list, iterates over it, and then calls a PIDF method? What's does the `.setPIDF()` method do? Ty

Reply:
The callback gets the list of existing motors and changes their PIDF values for closed-loop control. It exists because if we're in tuning mode, any changes made in SmartDashboard need to get sent to the hardware instantly.

Reply:
Ok, thanks so much! Yeah good implementation of async programming.